incubating/argocd-app-status/argocd_app_status.py

```diff
@@ -0,0 +1,100 @@
+from gql import Client, gql
+from gql.transport.requests import RequestsHTTPTransport
+import os
+import logging
+
+RUNTIME     = os.getenv('RUNTIME')
+APPLICATION = os.getenv('APPLICATION')
+
+CF_URL      = os.getenv('CF_URL', 'https://g.codefresh.io')
+CF_API_KEY  = os.getenv('CF_API_KEY')
+CF_STEP_NAME= os.getenv('CF_STEP_NAME', 'STEP_NAME')
+LOG_LEVEL   = os.getenv('LOG_LEVEL', "info")
+
+#######################################################################
+
+
+def main():
+    log_format = "%(asctime)s:%(levelname)s:%(name)s.%(funcName)s: %(message)s"
+    logging.basicConfig(format = log_format, level = LOG_LEVEL.upper())
+
+    get_app_status()
+
+    ## Generating link to the Apps Dashboard
+    CF_OUTPUT_URL_VAR = CF_STEP_NAME + '_CF_OUTPUT_URL'
+    link_to_app = get_link_to_apps_dashboard()
+    export_variable(CF_OUTPUT_URL_VAR, link_to_app)
+
+
+#######################################################################
+
+def get_query(query_name):
+    ## To do: get query content from a variable, failback to a file
+    with open('queries/'+query_name+'.graphql', 'r') as file:
+        query_content = file.read()
+    return gql(query_content)
+
+def get_runtime():
+    transport = RequestsHTTPTransport(
+        url = CF_URL + '/2.0/api/graphql',
+        headers={'authorization': CF_API_KEY},
+        verify=True,
+        retries=3,
+    )
+    client = Client(transport=transport, fetch_schema_from_transport=False)
+    query = get_query('getRuntime') ## gets gql query
+    variables = {
+        "runtime": RUNTIME
+    }
+    runtime = client.execute(query, variable_values=variables)
+    return runtime
+
+def get_runtime_ns():
+    runtime = get_runtime()
+    runtime_ns = runtime['runtime']['metadata']['namespace']
+    logging.debug("Runtime Namespace: %", runtime_ns)
+    return runtime_ns
+
+def get_link_to_apps_dashboard():
+    runtime_ns= get_runtime_ns()
+    url_to_app = CF_URL+'/2.0/applications-dashboard/'+RUNTIME+'/'+runtime_ns+'/'+APPLICATION+'/timeline'
+    return url_to_app
+
+def get_app_status():
+    runtime_api_endpoint = CF_URL + '/2.0/api/graphql'
+    transport = RequestsHTTPTransport(
+        url=runtime_api_endpoint,
+        headers={'authorization': CF_API_KEY},
```

Comment:
Just a suggestion here:
This var should be called `platform_api_endpoint` or something like that.

I feel `runtime_api_endpoint` makes reference to the API exposed by the app-proxy: `/app-proxy/api/graphql`

Just a comment

Reply:
Changed